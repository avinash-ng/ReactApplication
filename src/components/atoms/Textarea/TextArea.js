import React from 'react'
import { TextareaAutosize } from '@material-ui/core';
const TextArea = (props) => {
    return (
        <TextareaAutosize placeholder={props.placeholder} ></TextareaAutosize>
    )

}

export default TextArea;