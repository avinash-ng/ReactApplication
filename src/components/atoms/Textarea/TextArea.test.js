import TextArea from './TextArea'
import expect from 'expect'
import sinon from 'sinon';
import { shallow, configure } from 'enzyme'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
// import renderer from 'react-test-renderer'
import { render } from '@testing-library/react';
import { TextareaAutosize } from '@material-ui/core';

configure({ adapter: new Adapter() });
describe('<TextArea /> ', () => {
    it('renders TextArea element', () => {
        const RenderComponent = shallow(
            <TextArea ></TextArea>
        );
        // console.log(RenderComponent)
        expect(RenderComponent.exists()).toBe(true)
    })

    it('recieved props correctly', () => {
        const placeholder = "Enter your text here..."
        const RenderComponent = shallow(<TextArea placeholder={placeholder}></TextArea>)
        expect(RenderComponent.find(TextareaAutosize).prop('placeholder')).toEqual(placeholder)

    })


    it('Matches snapshot', () => {
            const placeholder = "Enter your text..."
            const RenderComponent = shallow(<TextArea placeholder={placeholder}></TextArea>)
            expect(RenderComponent).toMatchSnapshot()
    
    })
})