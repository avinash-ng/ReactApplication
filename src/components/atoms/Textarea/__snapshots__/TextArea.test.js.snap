// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea />  Matches snapshot 1`] = `ShallowWrapper {}`;
